sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.cs

```diff
@@ -41,4 +41,14 @@ public void Validate(object foobar)
             Do("foobar"); // Compliant - matches one of the parameter name
         }
     }
+
+    public struct Foo
+    {
+        public const string NameConst = "foobar"; // Noncompliant {{Define a constant instead of using this literal 'foobar' 4 times.}}
+//                                      ^^^^^^^^
+
+        private string name1 = "foobar"; // Secondary
+        private string name2 = "foobar"; // Secondary
+        private string name3 = "foobar"; // Secondary
+    }
```

Comment:
What about nested classes?
* If a nested class has too many duplicates, will we report two issues, once against the nested class and once against the containing class?
* should we report if e.g. the containing class has three duplicates and the nested class has two?

Reply:
Again, good catch! I'll report/analyze only from the top-level class, WDYT?

Reply:
It's not perfect, but it's probably simplest. We can always make it smarter later if necessary.

Reply:
...and I hesitate to say it, but... partial classes? (maybe just document that as a false negative)